File: TaskManager/src/UpdateIssueCommand.java
public class UpdateIssueCommand implements Command {
    private IssueRepository repository;
    private Issue issue;
    private String newTitle;
    private String newDescription;
    private String oldTitle;
    private String oldDescription;
    public UpdateIssueCommand(IssueRepository repository, Issue issue, String
            newTitle, String newDescription) {
        this.repository = repository;
        this.issue = issue;
        this.newTitle = newTitle;
        this.newDescription = newDescription;
    }
    @Override
    public void execute() {
        oldTitle = issue.getTitle();
        oldDescription = issue.getDescription();
        issue.setTitle(newTitle);
        issue.setDescription(newDescription);
    }
    @Override
    public void undo() {
        issue.setTitle(oldTitle);
        issue.setDescription(oldDescription);
    }
}

File: taskManager/labfinal_210042111/UpdateIssueCommand.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace labfinal_210042111
{
    public class UpdateIssueCommand
    {
        private IssueRepository repository;
        private Issue issue;
        public string newTitle;
        public string newDescription;
        public UpdateIssueCommand(IssueRepository repository, Issue issue, string newTitle, string newDescription)
        {
            this.repository = repository;
            this.issue = issue;
            this.newTitle = newTitle;
            this.newDescription = newDescription;
        }

        public void Execute()
        {

            issue.setTitle(newTitle);
            issue.setDescription(newDescription);

        }
        public void undo()
        {
            string title= issue.getTitle();
            string des = issue.getDescription();
            issue.undoTitle(title);
            issue.undoDescription(des);
            //traceback to previous version
        }

    }
}
